org.lflang/src/org/lflang/generator/CppGenerator.kt

```diff
@@ -0,0 +1,263 @@
+
+    /** Path to the preamble header file corresponding to this resource */
+    private val Resource.preambleHeaderPath get() = this.genDir.resolve("_lf_preamble.hh")
+
+    /** Path to the preamble source file corresponding to this resource */
+    private val Resource.preambleSourcePath get() = this.genDir.resolve("_lf_preamble.cc")
+
+    /** Path to the header file corresponding to this reactor */
+    private val Reactor.headerPath get() = this.eResource().genDir.resolve("${this.name}.hh")
+
+    /** Path to the implementation header file corresponding to this reactor (needed for generic reactors) */
+    private val Reactor.headerImplPath get() = this.eResource().genDir.resolve("${this.name}_impl.hh")
+
+    /** Path to the source file corresponding to this reactor (needed for non generic reactors)  */
+    private val Reactor.sourcPath get() = this.eResource().genDir.resolve("${this.name}.cc")
+
+    /** Convert a log level to a severity number understood by the reactor-cpp runtime. */
+    private val TargetProperty.LogLevel.severity
+        get() = when (this) {
+            TargetProperty.LogLevel.ERROR -> 1
+            TargetProperty.LogLevel.WARN -> 2
+            TargetProperty.LogLevel.INFO -> 3
+            TargetProperty.LogLevel.LOG -> 4
+            TargetProperty.LogLevel.DEBUG -> 4
+        }
+
+    override fun doGenerate(resource: Resource, fsa: IFileSystemAccess2, context: IGeneratorContext) {
+        super.doGenerate(resource, fsa, context)
+
+        // stop if there are any errors found in the program by doGenerate() in GeneratorBase
+        if (generatorErrorsOccurred) return
+
+        // abort if there is no main reactor
+        if (mainDef == null) {
+            println("WARNING: The given Lingua Franca program does not define a main reactor. Therefore, no code was generated.")
+            return
+        }
+
+        generateFiles(fsa)
+    }
+
+    private fun generateFiles(fsa: IFileSystemAccess2) {
+        val srcGenPath = fileConfig.srcGenPath
+        val relSrcGenPath = fileConfig.srcGenBasePath.relativize(srcGenPath)
+
+        // generate the cmake script
+        fsa.generateFile(relSrcGenPath.resolve("CMakeLists.txt").toString(), generateCmake())
+
+        // copy static library files over to the src-gen directory
+        val genIncludeDir = srcGenPath.resolve("__include__")
+        copyFileFromClassPath("${libDir}/lfutil.hh", genIncludeDir.resolve("lfutil.hh").toString())
+        copyFileFromClassPath("${libDir}/time_parser.hh", genIncludeDir.resolve("time_parser.hh").toString())
+        copyFileFromClassPath("${libDir}/3rd-party/CLI11.hpp", genIncludeDir.resolve("CLI").resolve("CLI11.hpp").toString())
+    }
+
+    @Suppress("LocalVariableName") // allows us to use capital S as variable name below
+    private fun generateCmake(): String {
+        val runtimeVersion = targetConfig.runtimeVersion ?: "26e6e641916924eae2e83bbf40cbc9b933414310"
```

Comment:
Where does this default version comes from ? Would be nice to put this in a named `const val` in the companion object